File: generated-code/python/codecraft/TcpReadWrite/model/build_properties.py
from model.entity_type import EntityType

class BuildProperties:
    """Entity's build properties"""

    def __init__(self, options, init_health):
        self.options = options
        """Valid new entity types"""
        self.init_health = init_health
        """Initial health of new entity. If absent, it will have full health"""

    @staticmethod
    def read_from(stream):
        """Read BuildProperties from input stream
        """
        options = []
        for _ in range(stream.read_int()):
            options_element = EntityType(stream.read_int())
            options.append(options_element)
        if stream.read_bool():
            init_health = stream.read_int()
        else:
            init_health = None
        return BuildProperties(options, init_health)

    def write_to(self, stream):
        """Write BuildProperties to output stream
        """
        stream.write_int(len(self.options))
        for element in self.options:
            stream.write_int(element)
        if self.init_health is None:
            stream.write_bool(False)
        else:
            stream.write_bool(True)
            stream.write_int(self.init_health)

    def __repr__(self):
        return "BuildProperties(" + \
            repr(self.options) + \
            ", " + \
            repr(self.init_health) + \
            ")"

File: generated-code/php/codecraft/TcpReadWrite/Model/BuildProperties.php
<?php

namespace Model {
    require_once 'Model/EntityType.php';
    require_once 'Stream.php';

    /**
     * Entity's build properties
     */
    class BuildProperties
    {
        /**
         * Valid new entity types
         */
        public array $options;
        /**
         * Initial health of new entity. If absent, it will have full health
         */
        public ?int $initHealth;
    
        function __construct(array $options, ?int $initHealth)
        {
            $this->options = $options;
            $this->initHealth = $initHealth;
        }
    
        /**
         * Read BuildProperties from input stream
         */
        public static function readFrom(\InputStream $stream): BuildProperties
        {
            $options = [];
            $optionsSize = $stream->readInt32();
            for ($optionsIndex = 0; $optionsIndex < $optionsSize; $optionsIndex++) {
                $optionsElement = \Model\EntityType::readFrom($stream);
                $options[] = $optionsElement;
            }
            if ($stream->readBool()) {
                $initHealth = $stream->readInt32();
            } else {
                $initHealth = NULL;
            }
            return new BuildProperties($options, $initHealth);
        }
        
        /**
         * Write BuildProperties to output stream
         */
        public function writeTo(\OutputStream $stream): void
        {
            $stream->writeInt32(count($this->options));
            foreach ($this->options as $element) {
                $stream->writeInt32($element);
            }
            if (is_null($this->initHealth)) {
                $stream->writeBool(false);
            } else {
                $stream->writeBool(true);
                $stream->writeInt32($this->initHealth);
            }
        }
    }
}